File: sqrt.c
#include <math.h>
#include <stdio.h>

double newton(double guess, double x, double tol) {
        return fabs(x - pow(guess, 2)) < tol
                ? guess
                : newton((guess + x / guess) / 2, x, tol);
}

int main(void) {
	double res = newton(1, 13, 0.000001);
	printf("%f\n%f\n", res, pow(res, 2));

	return 0;
}

File: sqrt.py
def newton(guess, x, tol):
    return guess if abs(x - guess**2) < tol\
            else newton((guess + x / guess) / 2, x, tol)

if __name__ == "__main__":
    res = newton(1, 13, 0.0000001)
    print("{}\n{}".format(res, res**2))
